ui_test/src/comments.rs

```diff
@@ -72,14 +109,45 @@ impl Comments {
 
     /// Parse comments in `content`.
     /// `path` is only used to emit diagnostics if parsing fails.
-    pub(crate) fn parse(path: &Path, content: &str) -> Self {
+    ///
+    /// This function will only parse `//@` and `//~` style comments
+    /// and ignore all others
+    fn parse_checked(path: &Path, content: &str) -> Self {
         let mut this = Self::default();
-        let error_pattern_regex =
-            Regex::new(r"//(\[(?P<revision>[^\]]+)\])?~(?P<offset>\||[\^]+)? *(?P<level>ERROR|HELP|WARN|NOTE)?:?(?P<text>.*)")
-                .unwrap();
 
         // The line that a `|` will refer to
         let mut fallthrough_to = None;
+        for (l, line) in content.lines().enumerate() {
+            let l = l + 1; // enumerate starts at 0, but line numbers start at 1
+            if let Some((_, command)) = line.split_once("//@") {
+                let command = command.trim();
+                if let Some((command, args)) = command.split_once(':') {
+                    this.parse_command_with_args(command, args, path, l);
+                } else if let Some((command, _comments)) = command.split_once(' ') {
+                    this.parse_command(command, path, l)
+                } else {
+                    this.parse_command(command, path, l)
+                }
+            } else if let Some((_, pattern)) = line.split_once("//~") {
+                this.parse_pattern(pattern, &mut fallthrough_to, path, l)
+            } else if let Some((_, pattern)) = line.split_once("//[") {
+                this.parse_revisioned_pattern(pattern, &mut fallthrough_to, path, l)
+            } else {
+                fallthrough_to = None;
+            }
+        }
+        this
+    }
+
+    /// Parse comments in `content`.
+    /// `path` is only used to emit diagnostics if parsing fails.
+    pub(crate) fn parse(path: &Path, content: &str) -> Self {
```

Comment:
I don't think we want to keep maintaining all this code.

> (or in fact, turn them into errors so that accidental usage of old-style comments will be detected)

If you want to do that, please just use a single regex like `// *(compiler?-flag|only-|ignore-)` and warn/error when that matches. We don't need the full parser for this.

Reply:
What's the issue with just porting them all now?

I don't like having so many dead code paths here.

Reply:
Mostly because it's very conflict prone and I'd have to rebase too often ^^